### DISCUSSION TITLE
Kernel size on Spline Conv?

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#4417] [Q&A] [answered]
Hi Matt, 
I have two questions. 

1. What does the ```kernel_size``` mean here ```self.conv1 = SplineConv(1, 64, dim=2, kernel_size=5)```?
I don't see any detailed explanation on the [SplineCNN paper](https://openaccess.thecvf.com/content_cvpr_2018/html/Fey_SplineCNN_Fast_Geometric_CVPR_2018_paper.html)
Is it similar to ```k``` in k-hop embedding?

2. I also have a question about the mechanism of mini-batch training in ```SplineCNN```. 
I think it should be a little bit different from typical GNNs. In a typical GNN, ```M``` nodes are selected, and node embeddings are calculated for mini-batch training right? 
Then how does it work for ```SplineCNN```? (Mini-batch handling part is a little bit confusing to me :<)

**Comment by rusty1s** [comment, marked as answer]
1. The `kernel_size` means how many control points/parameters there will be in each dimension. A `kernel_size=5` across two dimensions will utilize 5x5=25 parameters per kernel (similar to a 5x5 kernel in CNNs).
2. We haven't applied SplineCNN in large-graphs with sampling techniques yet, but it should be similar to apply than any other GNN layer in PyG. Note that we distinguish between mini-batching of single graphs (in graph datasets) and mini-batching of sub-graphs (in single large graphs). I can try to explain it in more detail in case you specify your issues more concretely :)
